README.md

```diff
@@ -2,3 +2,163 @@
 
 An initial scalable filesystem for Linux operating systems
 
+- [Why use initoverlayfs to generate initramfs images?](#why-use-initoverlayfs-to-generate-initramfs-images-)
+- [Installation](#installation)
+    * [Step 1 - Deploy the software](#step-1---deploy-the-software)
+    * [Step 2 - Run initoverlayfs-install](#step-2---run-initoverlayfs-install)
+    * [Step 3 - Reboot to test](#step-3---reboot-to-test)
+    * [Step 4 - Validating the boot](#step-4---validating-the-boot)
+
+# Why use initoverlayfs to generate initramfs images?
+
+An initramfs (Initial RAM File System) image is a fundamental component in preparing Linux systems during the boot process, preceding the initiation of the init process. 
+
+Typically, generating an initramfs involves assembling all available kernel modules and necessary files to boot and support any hardware using the specific Linux kernel version XYZ.  
+
+However, this conventional approach presents a significant challenge: loading such a voluminous image into memory during boot is time-consuming and can be problematic in critical scenarios such as healthcare, automotive, and aviation.
+
+Conversely, the initoverlayfs approach proposes a solution: "**dividing the initramfs image into two parts**." 
+
+This division entails segregating the initramfs image into two distinct components.
+The first component houses only the kernel modules necessary for storage, responsible for bringing up the operating system. Subsequently, it triggers the second component, containing all additional kernel modules and essential files required to support the Linux boot process. 
```

Comment:
Second part of this should be 
The second componant is mounted by the first comportment ...

Reply:
I would say "The first component (initramfs) houses only the kernel modules and udev-rules necessary for storage, responsible for bringing up storage device quickly. Subsequently, it mounts and switches the second component (initoverlayfs), containing all additional kernel modules and essential files required to support the Linux boot process."

The first part does storage and nothing else. There's actually a third component rootfs, but nothing there changes I guess. Some kernel modules are in rootfs and not initoverlayfs, just as some kernel modules will be in initoverlayfs and not initramfs.

There is a gluttony of essential files put in initoverlayfs that would have been in initramfs in the past, such as ostree tools, decryption tools, filesystem verification tools, bootsplash tools, it's 1,000s of files, but "essential files" is good enough without delving into the detail like you did. :+1: